plugins/src/ios/OptimoveSDKPlugin.swift

```diff
@@ -34,4 +34,105 @@ import NotificationCenter
         guard let config = OptimoveSDKPlugin.config else { return }
         Optimove.initialize(with: config)
     }
+    
+    @objc(reportEvent:)
+    func reportEvent(command: CDVInvokedUrlCommand) {
+        var name = ""
+        var parameters: [String: Any] = [:]
+        
+        if let n = command.arguments[0] as? String {
+            name = n
+        }
+        
+        if let param = command.arguments[1] as? [String: Any] {
+            parameters = param
+        }
+        
+        Optimove.shared.reportEvent(name: name, parameters: parameters)
+        
+        self.commandDelegate.run {
+            let pluginResult = CDVPluginResult(status: .ok)
+            self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
+        }
+    }
+    
+    @objc(reportScreenVisit:)
+    func reportScreenVisit(command: CDVInvokedUrlCommand) {
+        guard let screenTitle = command.arguments.first as? String else {
+            let pluginResult = CDVPluginResult(status: .error, messageAs: "screenTitle is invalid")
```

Comment:
In general, can we just check the implementation of validation and error handling/error messages against the android implementation so they are consistent for each method?

Reply:
OK perhaps you can sync up with Tom about this to go over it with him

Reply:
will do

Reply:
validation and error handling/error messages will be handled in Optimove.js file